src/test/java/net/minecraftforge/debug/StructureGenTest.java

```diff
@@ -173,18 +182,18 @@ public boolean isInsideStructure(BlockPos pos)
                 logger.info("Checking for structure {}", generator);
                 isAt = false;
             }
-            return pos.getX() /16 == 4 && pos.getZ() /16 == 4;
+            return pos.getX() / 16 == 4 && pos.getZ() / 16 == 4;
         }
 
         @Override
-        public List<Biome.SpawnListEntry> getSpawns(World world, EnumCreatureType creatureType, BlockPos pos)
+        public List<Biome.SpawnListEntry> getSpawns(List<Biome.SpawnListEntry> defaults, World world, EnumCreatureType creatureType, BlockPos pos)
```

Comment:
You don't need to return a list any more; modification can be done within the list provided.

Reply:
really?

```
defaults.clear();
defaults.addAll(myNeverChangingFinalStaticList);
```

all the time?

Reply:
What if the user returns an immutable list?

Reply:
ok

Reply:
Yes, after implementing these changes and adding entity spawns to it, I agree with @HenryLoenwind .
It might be a nicer solution, if you don't change the spawn list every time, but since you most certainly always want to return your own list if you are implementing this interface.

Not sure about the following scenario though:
You want to add additional entries to the biome's ones.

With liach's approach you could just add them to the list. If we return a list you would have to create a new list to join the passed default list (which must not be modified) and your own final list.
Wouldn't make a difference in the background though as we  always have to create a new list in liach's approach anyway.

Reply:
Any other opinions on this or should I just revert the change?
@liach

Reply:
reverted for now

Reply:
I originally thought both are O(N) and have the same amortized time, it is okay to be like this. Do we allow modifying the default list? If yes, ask users to return a copied new list every time.

Reply:
The default list must not be modified because it is the list stored in the biome object
Therefore I am passing an unmodifiable list as default list here https://github.com/MinecraftForge/MinecraftForge/pull/4608/files#diff-a34fd8258a2226b73fa9f9405d45f27eR123

This is indirectly a change to the IChunkGenerator because in vanilla it always returns a modifiable list (as far as I can tell), but it definitively should not be modified so I think this is less of an issue. We could always ask for a copy of course, but I don't think that is necessary.